### DISCUSSION TITLE
How to support socket.io in NEXTJS14

### DISCUSSION THREAD

**Discussion by MasterProvider** [vercel/next.js#59454] [Help]
### Summary

Any approach for this since the response is not available in App route api

/app/api/socket/route.js

```
import { NextResponse } from "next/server";
import { Server } from "socket.io";


export const GET = async () => {
    let res = NextResponse.next()

    if (res.socket.server.io) {
      return res;
    }

    const io = new Server(res.socket.server,{ path:'/api/socket', addTrailingSlash: false});
    res.socket.server.io = io;

    io.on("connection", (socket) => {
      console.log('A user connected');

      socket.on("send-message", (obj) => {
        io.emit("receive-message", obj);
      });

      socket.on('disconnect', () => {
        console.log('A user disconnected');
      });
      
    });

    return res;
};

```

**Comment by Mohammad-Shahid-07** [comment]
Should you discover an answer to the problem at hand, kindly share it.

**Reply by MasterProvider** [reply]
still not supported even in 14.1.0

**Comment by manusaavedra** [comment]
https://medium.com/@farmaan30327/implement-web-sockets-with-nextjs-and-api-routes-6a14916e6c1a

This seems to be a solution, it would be to create a custom server, change in the scripts of the package.json so that the app launches from your new server.

**Reply by YuriyBishko** [reply]
It doesn't work in production, only in development 😢
